invenio_rdm_records/records/systemfields/access/grants.py

```diff
@@ -86,22 +99,23 @@ def subject_id(self):
 
         return self.subject.id
 
-    def covers(self, permission):
-        """Check if this Grant covers the specified permission."""
-        # TODO check this via a permission hierarchy
-        return True
-
     def to_need(self):
-        """Create the need that this grant provides."""
+        """Create the need that this grant provides.
+
+        Note: This operation doesn't resolve the subject, and thus won't cause any
+        database lookups even if the. On the flipside, it also won't check if the
+        subject exists.
+        """
         if self.subject_type == "user":
-            need = UserNeed(self.subject.id)
+            # NOTE: the `UserNeed` must have an integer as ID, otherwise it won't match
+            need = UserNeed(int(self.subject_id))
```

Comment:
in contrast to e.g. the `datastore`, we _need_ the user's ID to be numeric in the `UserNeed` – otherwise, the permissions will never be granted